TransIT.BLL/Services/UserService.cs

```diff
@@ -0,0 +1,128 @@
+            IUserRepository userRepository,
+            IRoleRepository roleRepository)
+        {
+            _uow = uow;
+            _userRepository = userRepository;
+            _roleRepository = roleRepository;
+            _hasher = hasher;
+        }
+
+        /// <summary>
+        /// Finds user by id
+        /// </summary>
+        /// <param name="userId">Id of user</param>
+        /// <returns>User or null unless not found</returns>
+        public Task<User> Get(int userId) =>
+            _userRepository.GetByIdAsync(userId);
+
+        /// <summary>
+        /// Gives users with pagination
+        /// </summary>
+        /// <param name="index">Where to start</param>
+        /// <param name="amount">Amount to give</param>
+        /// <returns>List of users</returns>
+        public Task<IEnumerable<User>> Get(uint index, uint amount) =>
+            _userRepository.GetRangeAsync(index, amount);
+
+        /// <summary>
+        /// Creates user if login and password not empty and does not exist in DB
+        /// hashes password and set zero to id
+        /// </summary>
+        /// <param name="value">User model</param>
+        /// <returns>Is successful</returns>
+        public async Task<bool> Create(User value)
+        {
+            if (string.IsNullOrEmpty(value.Login)
+                || string.IsNullOrEmpty(value.Password)
+                || (await _userRepository.GetAllAsync(u =>
+                u.Login == value.Login)).Any()) 
+                return false;
+            
+            var role = value.Role.Name.ToUpper();
+            value.Role = (await _roleRepository
+                .GetAllAsync(r => r.Name == role))
+                .SingleOrDefault();
+
+            if (value.Role == null) return false;
+
+            value.Id = 0;
+            value.Password = _hasher.HashPassword(value.Password);
+            
+            await _userRepository.AddAsync(value);
+            await _uow.SaveAsync();
+            return true;
+        }
+
+        /// <summary>
+        /// Updates user if it is found
+        /// </summary>
+        /// <param name="value">User to update</param>
+        /// <returns>Is successful</returns>
+        public async Task<bool> Update(User value)
+        {
+            if (await _userRepository.GetByIdAsync(value.Id) == null)
+                return false;
+
+            _userRepository.Update(value);
+            await _uow.SaveAsync();
+            return true;
+        }
+
+        /// <summary>
+        /// Removes a user in DB
+        /// </summary>
+        /// <param name="userId">Id of user to delete</param>
+        /// <returns>Is successful</returns>
+        public async Task<bool> Delete(int userId)
+        {
+            var user = await _userRepository.GetByIdAsync(userId);
+            if (user == null) return false;
+            
+            _userRepository.Remove(user);
+            await _uow.SaveAsync();
+            return true;
```

Comment:
Select and then delete...
This can be done with one query (delete), but we need Remove method on repository to receive ID

Reply:
Try catch